components/js_lang/js_lang_core.h

```diff
@@ -398,6 +466,18 @@ class Js_thd {
 
   // Map with per user-account contexts for the connection.
   std::unordered_map<std::string, Auth_id_context> m_auth_id_contexts;
+
+  // Flag indicating that current connection is running JS code which
+  // requires call to Isolate::TerminateExecution() if connection/query
+  // is KILLed, as well as pointing to the specific Isolate used for this.
+  v8::Isolate *m_killable_isolate;
+  // Lock protecting m_killable_isolate member.
+  std::mutex m_killable_isolate_lock;
+
+  // MYSQL_THD -> Js_thd map which enables thread-safe lookup of Js_thd
+  // context by KILL implementation and lock protecting it.
+  static std::unordered_map<MYSQL_THD, Js_thd *> s_thd_to_js_thd_map;
+  static std::mutex s_thd_to_js_thd_map_lock;
```

Comment:
Does it make sense to change this to `std::shared_mutex` instead?

Reply:
Hmm... I guess such a change doesn't make much sense if we are talking about KILL statements only. They are rare, so shared_mutex would be acquired in exclusive mode most of the times in this case anyway. But in for scenarios when MAX_EXECUTION_TIME limit is exceeded often it might be a different story...

In any case this code is gone after the second commit :)